### PR TITLE
refactor: remove jquery from application

### PR DESCRIPTION
**Pull request by przemvs**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: src/script/util/DebugUtil.ts [hunk 1]
```diff
@@ -77,8 +77,6 @@ export class DebugUtil {
   /** Used by QA test automation. */
   public readonly userRepository: UserRepository;
   /** Used by QA test automation. */
-  public readonly $: JQueryStatic;
-  /** Used by QA test automation. */
```
Comment (atomrc) [inline src/script/util/DebugUtil.ts:81]: we should restore this as it will break automation

File: src/script/util/DebugUtil.ts [hunk 2]
```diff
@@ -89,7 +87,6 @@ export class DebugUtil {
     private readonly callState = container.resolve(CallState),
     private readonly core = container.resolve(Core),
   ) {
-    this.$ = $;
```
Comment (atomrc) [inline src/script/util/DebugUtil.ts]: and this

File: src/script/main/app.ts [hunk 3]
```diff
@@ -914,7 +912,7 @@ class App {
 // Setting up the App
 //##############################################################################
 
-$(async () => {
+(async () => {
```
Comment (atomrc) [inline src/script/main/app.ts:915]: Does this work? Cause it's not the equivalent of `$(...)`. 
If you tell me it works, (the app loads) then we are good

File: package.json [hunk 4]
```diff
@@ -17,8 +17,6 @@
     "eslint-plugin-header": "3.1.1",
     "highlight.js": "11.6.0",
     "http-status-codes": "2.2.0",
-    "jquery": "3.6.1",
```
Comment (atomrc) [inline package.json:21]: we might need to restore this for ko

File: src/script/util/util.ts [hunk 5]
```diff
@@ -404,3 +404,15 @@ export const getSelectionPosition = (element: HTMLTextAreaElement, currentMentio
...
+
+export const removeAnimationsClass = (element: HTMLElement | null) => {
+  if (element) {
+    element.addEventListener('animationend', () => {
+      if (element.classList.contains(incomingCssClass)) {
+        element.classList.remove(incomingCssClass);
+      }
+    });
+  }
+};
```
Comment (atomrc) [inline src/script/util/util.ts:418]: super nice 🤩

File: src/script/main/app.ts (hunk 3 above)
Comment (przemvs) [reply src/script/main/app.ts:915]: app renders :D

File: src/demo/demo.html [hunk 6]
```diff
@@ -145,27 +145,39 @@ <h1>Emoji range covered by 'emoji' fontface</h1>
     <div id="emojis"></div>
 
     <script>
-      $(() => {
+      (() => {
```
Comment (atomrc) [inline src/demo/demo.html]: This file can be safely deleted now 😉

File: src/script/components/Modals/UserModal/UserModal.tsx [hunk 7]
```diff
@@ -55,7 +54,7 @@ const brandName = Config.getConfig().BRAND_NAME;
 
 const UserModal: React.FC<UserModalProps> = ({
   userId,
-  onClose = noop,
+  onClose = () => undefined,
```
Comment (atomrc) [inline src/script/components/Modals/UserModal/UserModal.tsx]:
```suggestion
  onClose = () => {},
```
Even shorter ;)

**Review by aweiss-dev — APPROVED** [review]

**Review by atomrc — APPROVED** [review]

**Pull request merged**